staging/publishing/rules.yaml

```diff
@@ -669,64 +669,6 @@ rules:
         branch: release-1.14
       - repository: client-go
         branch: release-11.0
-- destination: csi-api
```

Comment:
I hope this is fine

Reply:
needs rebasing, this file now only contains the rules for the master branch, so removing for 1.15 would be fine